tldw_Server_API/app/core/DB_Management/Kanban_DB.py

```
@@ -122,30 +186,65 @@ def __init__(self, db_path: str, user_id: str) -> None:
-        conn = sqlite3.connect(self.db_path, timeout=30, isolation_level=None)
+    def _configure_connection(self, conn: sqlite3.Connection, *, enable_wal: bool = True) -> None:
         conn.row_factory = sqlite3.Row
+        # Foreign keys and timeout are required for integrity and responsiveness.
+        for statement in ("PRAGMA foreign_keys=ON", "PRAGMA busy_timeout=30000"):
+            try:
+                conn.execute(statement)
+            except Exception as e:
+                logger.error(f"Failed to set critical PRAGMA option {statement}: {e}")
+                raise KanbanDBError(
+                    f"Failed to set critical PRAGMA option {statement}: {e}"
+                ) from e
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)